packages/astro/src/runtime/client/dev-overlay/entrypoint.ts

```diff
@@ -0,0 +1,93 @@
+import type { DevOverlayPlugin as DevOverlayPluginDefinition } from '../../../@types/astro.js';
+import { type AstroDevOverlay, type DevOverlayPlugin } from './overlay.js';
+
+let overlay: AstroDevOverlay;
+
+document.addEventListener('DOMContentLoaded', async () => {
+	const [
+		{ loadDevOverlayPlugins },
+		{ default: astroDevToolPlugin },
+		{ default: astroAuditPlugin },
+		{ default: astroXrayPlugin },
+		{ AstroDevOverlay, DevOverlayCanvas },
+		{ DevOverlayCard },
+		{ DevOverlayHighlight },
+		{ DevOverlayTooltip },
+		{ DevOverlayWindow },
+	] = await Promise.all([
+		// @ts-expect-error
+		import('astro:dev-overlay'),
+		import('./plugins/astro.js'),
+		import('./plugins/audit.js'),
+		import('./plugins/xray.js'),
+		import('./overlay.js'),
+		import('./ui-library/card.js'),
+		import('./ui-library/highlight.js'),
+		import('./ui-library/tooltip.js'),
+		import('./ui-library/window.js'),
+	]);
```

Comment:
Moved the imports into the `DOMContentLoaded`, that way we really do the big things outside of the page load